testutils/clustermanager/gke.go

```diff
@@ -299,65 +222,39 @@ func (gc *GKECluster) Acquire() error {
 	for i, region := range regions {
 		// Restore innocence
 		err = nil
-		rb := &container.CreateClusterRequest{
-			Cluster: &container.Cluster{
-				NodePools: []*container.NodePool{
-					{
-						Name: "default-pool",
-						Autoscaling: &container.NodePoolAutoscaling{
-							Enabled:      true,
-							MinNodeCount: gc.Request.MinNodes,
-							MaxNodeCount: gc.Request.MaxNodes,
-						},
-						Config: &container.NodeConfig{
-							MachineType: gc.Request.NodeType,
-						},
-					},
-				},
-				Name: clusterName,
-				// The default cluster version is not latest, has to explicitly
-				// set it as "latest"
-				InitialClusterVersion: defaultGKEVersion,
-				// Installing addons after cluster creation takes at least 5
-				// minutes, so install addons as part of cluster creation, which
-				// doesn't seem to add much time on top of cluster creation
-				AddonsConfig: gc.getAddonsConfig(),
-				// Equivalent to --enable-basic-auth, so that user:pass can be
-				// later on retrieved for setting up cluster roles. Use the
-				// default username from gcloud command, the password will be
-				// automatically generated by GKE SDK
-				MasterAuth: &container.MasterAuth{Username: "admin"},
-			},
-			ProjectId: *gc.Project,
+		var rb *container.CreateClusterRequest
+		rb, err = gke.NewCreateClusterRequest(&gc.Request.Request)
+		if err != nil {
+			break
 		}
 
-		clusterLoc := getClusterLocation(region, gc.Request.Zone)
+		clusterLoc := gke.GetClusterLocation(region, gc.Request.Zone)
 
 		// Deleting cluster if it already exists
-		existingCluster, _ := gc.operations.get(*gc.Project, clusterLoc, clusterName)
+		existingCluster, _ := gc.operations.GetCluster(*gc.Project, clusterLoc, clusterName)
 		if existingCluster != nil {
 			log.Printf("Cluster %q already exists in %q. Deleting...", clusterName, clusterLoc)
-			op, err = gc.operations.delete(*gc.Project, clusterLoc, clusterName)
+			op, err = gc.operations.DeleteCluster(*gc.Project, clusterLoc, clusterName)
 			if err == nil {
-				err = gc.wait(clusterLoc, op.Name, deletionTimeout)
+				err = gke.WaitUntilDone(gc.operations, *gc.Project, clusterLoc, op.Name, gke.DeletionTimeout)
 			}
 		}
 		// Creating cluster only if previous step succeeded
 		if err == nil {
 			log.Printf("Creating cluster %q in %q with:\n%+v", clusterName, clusterLoc, gc.Request)
-			op, err = gc.operations.create(*gc.Project, clusterLoc, rb)
+			op, err = gc.operations.CreateCluster(*gc.Project, clusterLoc, rb)
 			if err == nil {
-				err = gc.wait(clusterLoc, op.Name, creationTimeout)
+				err = gke.WaitUntilDone(gc.operations, *gc.Project, clusterLoc, op.Name, gke.CreationTimeout)
```

Comment:
this double reference on same line is tough! Can we have a wrapper in `gke` looks like this?
```
type GKEWrapper struct {
client SDKOperations
}
func (gw *GKEWrapper) WaitUntilDone() {}
```
Just an idea, the name of the wrapper could be something make more sense

Reply:
Moved the function to be internal, so there is no such problem anymore.